riscv-executor/src/lib.rs

```diff
@@ -1021,23 +1083,35 @@ impl<'a, 'b, F: FieldElement> Executor<'a, 'b, F> {
     }
 
     /// read register value, updating the register memory machine
-    fn reg_read(&mut self, step_offset: u32, reg: u32, selector_idx: u32) -> Elem<F> {
+    fn reg_read(&mut self, step_offset: u32, reg: u32, selector: u8) -> Elem<F> {
         let val = self.proc.get_reg_mem(reg);
-        if let ExecMode::Trace = self.mode {
-            self.proc
-                .regs_machine
-                .read(self.step + step_offset, reg, val, selector_idx);
-        }
+        self.proc.submachine_op(
```

Comment:
only trace mode or always needed?

Reply:
submachine_op is always only in trace mode, so the test is done inside it (cleaner code and less possibility of mistakes)